topeft/modules/corrections.py

```diff
@@ -937,31 +1054,61 @@ def ApplyJetCorrections(year, corr_type):
-        "* * " + topcoffea_path('data/JER/%s_MC_PtResolution_AK4PFchs.jr.txt' % jer_tag),
-        "* * " + topcoffea_path('data/JEC/Summer19UL%s_MC_L1FastJet_AK4PFchs.txt' % jec_tag),
-        "* * " + topcoffea_path('data/JEC/Summer19UL%s_MC_L2Relative_AK4PFchs.txt' % jec_tag),
-        "* * " + topcoffea_path('data/JEC/Quad_Summer19UL%s_MC_UncertaintySources_AK4PFchs.junc.txt' % jec_tag)
-    ])
-    jec_types = [
-        'FlavorQCD', 'FlavorPureBottom', 'FlavorPureQuark', 'FlavorPureGluon', 'FlavorPureCharm',
-        'BBEC1', 'Absolute', 'RelativeBal', 'RelativeSample'
-    ]
-    jec_regroup = ["Quad_Summer19UL%s_MC_UncertaintySources_AK4PFchs_%s" % (jec_tag,jec_type) for jec_type in jec_types]
-    jec_names = [
-        "%s_MC_SF_AK4PFchs" % jer_tag,
-        "%s_MC_PtResolution_AK4PFchs" % jer_tag,
-        "Summer19UL%s_MC_L1FastJet_AK4PFchs" % jec_tag,
-        "Summer19UL%s_MC_L2Relative_AK4PFchs" % jec_tag
-    ]
-    jec_names.extend(jec_regroup)
-    extJEC.finalize()
-    JECevaluator = extJEC.make_evaluator()
-    jec_inputs = {name: JECevaluator[name] for name in jec_names}
-    jec_stack = JECStack(jec_inputs)
-    name_map = jec_stack.blank_name_map
+    jec_year = year
+    if year.startswith("2016"):
+        jec_year = "2016preVFP" if year == "2016APV" else "2016postVFP"
+    if int(year.replace("APV", "")) < 2020:
+        jec_year += "_UL"
+
+    jet_algo = "AK4PFchs"
+    #jet_algo = "AK8PFPuppi"
+    if usejecstack:
+        extJEC = lookup_tools.extractor()
+        extJEC.add_weight_sets([
+            "* * " + topcoffea_path(f'data/JER/{jer_tag}_MC_SF_{jet_algo}.jersf.txt'),
+            "* * " + topcoffea_path(f'data/JER/{jer_tag}_MC_PtResolution_{jet_algo}.jr.txt'),
+            "* * " + topcoffea_path(f'data/JEC/Summer19UL{jec_tag}_MC_L1FastJet_{jet_algo}.txt'),
+            "* * " + topcoffea_path(f'data/JEC/Summer19UL{jec_tag}_MC_L2Relative_{jet_algo}.txt'),
+            "* * " + topcoffea_path(f'data/JEC/Quad_Summer19UL{jec_tag}_MC_UncertaintySources_{jet_algo}.junc.txt')
+        ])
+        jec_types = [
+            'FlavorQCD', 'FlavorPureBottom', 'FlavorPureQuark', 'FlavorPureGluon', 'FlavorPureCharm',
+            'BBEC1', 'Absolute', 'RelativeBal', 'RelativeSample'
+        ]
+        jec_regroup = ["Quad_Summer19UL%s_MC_UncertaintySources_{jet_algo}_%s" % (jec_tag,jec_type) for jec_type in jec_types]
+        jec_names = [
+            f"{jer_tag}_MC_SF_{jet_algo}",
+            f"{jer_tag}_MC_PtResolution_{jet_algo}",
+            f"Summer19UL{jec_tag}_MC_L1FastJet_{jet_algo}",
+            f"Summer19UL{jec_tag}_MC_L2Relative_{jet_algo}",
+        ]
+        jec_names.extend(jec_regroup)
+
+        extJEC.finalize()
+        JECevaluator = extJEC.make_evaluator()
+        jec_inputs = {name: JECevaluator[name.replace("Regrouped_", "")] for name in jec_names}
+        jec_stack = JECStack(jec_inputs)
+
+    elif useclib:
```

Comment:
Right now these `useclib` and `usejecstack` are opposites, so this could just be an else statement. Maybe it will change if you don't want `usejecstack` to _just_ be `not useclib`.

Reply:
I think that for the sake of clarity, the proposed approach is clearer, and also one can disentangle the two booleans when running any comparisons between the two approaches

Reply:
That's a fair point.